jetty-openid/src/test/java/org/eclipse/jetty/security/openid/JwtDecoderTest.java

```diff
@@ -0,0 +1,99 @@
+//
+//  ========================================================================
+//  Copyright (c) 1995-2019 Mort Bay Consulting Pty. Ltd.
+//  ------------------------------------------------------------------------
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.security.openid;
+
+import java.util.Map;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class JwtDecoderTest
+{
+    public static Stream<Arguments> paddingExamples()
+    {
+        return Stream.of(
+            Arguments.of("XXXX", "XXXX"),
+            Arguments.of("XXX", "XXX="),
+            Arguments.of("XX", "XX=="),
+            Arguments.of("XXX=", "XXX="),
+            Arguments.of("X-X", "X-X="),
+            Arguments.of("@#", "@#=="),
+            Arguments.of("X=", "X="),
+            Arguments.of("XX=", "XX="),
+            Arguments.of("XX==", "XX=="),
+            Arguments.of("XXX=", "XXX="),
+            Arguments.of("", "")
```

Comment:
These don't seem like valid Base64.

Reply:
Can we see valid `.decode()` examples being tested?
This whole padding thing seems way off (there's so many modes in the java Base64, one of them has got to handle this without the strange `padJWTSection()` hack)..
The use of the a `UrlDecoder` is also suspect (to me).
Still don't see a real world scenario / testcase that demonstrates the need for this padding concept (vs just using the correct Base64 decoder setup/configuration from the get go)

Reply:
`padJWTSection()` only adds the required padding to a string, it does not verify that the contents of the string are valid base64. I would have done more tests for `JwtDecoder.decode()` itself, but its hard to find example JWT strings without revealing personal information in them, which is why I was just generating a fake JWT.


As for why `padJWTSection` is necessary, is because the padding is manually removed when creating the JWS, see in [RFC7515](https://tools.ietf.org/html/rfc7515#section-2) it says:
```
Base64 encoding using the URL- and filename-safe character set
defined in Section 5 of RFC 4648 [RFC4648], with all trailing '='
characters omitted
```
and then gives an example of how to do the URL safe decoding and addition of the padding.
https://tools.ietf.org/html/rfc7515#appendix-C